File: eecs268/2020summer/video-lectures/2020.07.10.BST-intro/BST/BST.cpp
//public search
template <typename ItemType, typename KeyType>
ItemType BST<ItemType, KeyType>::search(KeyType key) const
{
	return( search(key, m_root) );
}

//private
template <typename ItemType, typename KeyType>
ItemType BST<ItemType, KeyType>::search(KeyType key, BNode<ItemType>* subtree) const
{
	//Board work: finish this definition
	//Assume that ItemType == KeyType
	//				ItemType < KeyType
	//				ItemType > KeyType
	//			are all defined
	
}

File: eecs268/2022summer/2022.07.19.bst-searching/bst.py
#bst.py

from bnode import BNode


class BST:
    def __init__(self):
        self._root = None

    def search(self, search_key):
        return(self._rec_search(search_key, self._root))

    def _rec_search(self, search_key, cur_node):
        #Board work
        #Raise a RuntimeError if search fails
